### DISCUSSION TITLE
CacheTime dynamic and infinity

### DISCUSSION THREAD

**Discussion by olup** [TanStack/query#586] [General]
Preparing for offline first persistence I was wondering if I could change the query cacheTime dynamically, depending on the network connection. If phone is online cached queries can be garbage collected every x minutes, but if I loose connectivity, I want to keep my queries around. Also and in the same use case, can I set cachetime to infinity ?

**Comment by tannerlinsley** [comment]
Yes you can. You can update it globally for the entire config at will and it will take effect for new or existing queries that render/rerender after the change. You can set it to infinity as well and it will never garbage collect :)
